utils.go

```diff
@@ -15,23 +44,139 @@ type Storage struct {
 
 // String implements Storager.String
 func (s *Storage) String() string {
-	panic("implement me")
+	return fmt.Sprintf("Storager us3 {Name: %s, WorkDir: %s", s.bucket, s.workDir)
+}
+
+func New(pairs ...types.Pair) (types.Servicer, types.Storager, error) {
+	return newServicerAndStorager(pairs...)
+}
+
+func NewServicer(pairs ...types.Pair) (types.Servicer, error) {
+	return newServicer(pairs...)
 }
 
 // NewStorager will create Storager only.
 func NewStorager(pairs ...types.Pair) (types.Storager, error) {
-	return newStorager(pairs...)
+	_, store, err := newServicerAndStorager(pairs...)
+	return store, err
 }
 
-// newStorager will create a storage client.
-func newStorager(pairs ...types.Pair) (store *Storage, err error) {
+func newServicer(pairs ...types.Pair) (srv *Service, err error) {
 	defer func() {
 		if err != nil {
-			err = services.InitError{Op: "new_storager", Type: Type, Err: formatError(err), Pairs: pairs}
+			err = services.InitError{
+				Op:    "new_servicer",
+				Type:  Type,
+				Err:   formatError(err),
+				Pairs: pairs,
+			}
 		}
 	}()
 
-	panic("implement me")
+	srv = &Service{}
+
+	opt, err := parsePairServiceNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	cp, err := credential.Parse(opt.Credential)
+	if err != nil {
+		return nil, err
+	}
+	if cp.Protocol() != credential.ProtocolHmac {
+		return nil, services.PairUnsupportedError{Pair: ps.WithCredential(opt.Credential)}
+	}
+	ak, sk := cp.Hmac()
+
+	ep, err := endpoint.Parse(opt.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var url string
+	switch ep.Protocol() {
+	case endpoint.ProtocolHTTP:
+		url, _, _ = ep.HTTP()
+	case endpoint.ProtocolHTTPS:
+		url, _, _ = ep.HTTPS()
+	default:
+		return nil, services.PairUnsupportedError{Pair: ps.WithEndpoint(opt.Endpoint)}
+	}
+
+	config := &us3.Config{
+		PublicKey:  ak,
+		PrivateKey: sk,
+		FileHost:   url,
+		BucketHost: "api.ucloud.cn",
+	}
+
+	srv.service, err = us3.NewFileRequest(config, nil)
```

Comment:
Maybe we need `NewBucketRequest` here, ref: https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/example/demo_bucket.go?

Reply:
I think it should be `NewFileRequest`. Although they share the same model, us3 requires `NewBucketRequest` not to call file-related APIs. Like the [US3](https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/request.go#:~:text=//NewBucketRequest%20%E5%88%9B%E5%BB%BA%E4%B8%80%E4%B8%AA,%E7%9B%B8%E5%85%B3%E6%96%87%E6%A1%A3%E3%80%82) comments says.

Reply:
Maybe we should separate the creation of `servicer` and `storager`, I don't think they can share a client.

Reply:
> I think it should be `NewFileRequest`. Although they share the same model, us3 requires `NewBucketRequest` not to call file-related APIs. 

`Servicer` is the container for storage. It is used to manage the bucket here in my view and just as `NewBucketRequest` for bucket related options.